scripts/src/modules/wikidata-overlays/wikidata-overlays.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * fetchData
+ *
+ * @param {String} url
+ * @returns {Promise<Object>}
+ *
+ * Fetch data from a given URL
+ */
+const fetchData = async (url) => {
+    const response = await fetch(url);
+
+    if (!response.ok) {
+        throw new Error("Network response was not ok");
+    }
+
+    return response.json();
+};
+
+const createTemplate = (data, tagType) => {
+    const template = document.createElement("template");
+
+    const tagTypeTemplate = tagType
+        ? `<p class="wikidata__description">${tagType}</p>`
+        : "";
+    const imageTemplate = data.imagePath
+        ? `<img src="${data.imagePath}" alt="" width="74" class="wikidata__header-img" loading="lazy">`
+        : "";
+    const countryTemplate = data.countryLabel
+        ? `<p class="wikidata__header-meta">${data.countryLabel}</p>`
+        : "";
+    const titleTemplate = data.label ? `<h3>${data.label}</h3>` : "";
+    const descriptionTemplate = data.description
+        ? `<p>${data.description}</p>`
+        : "";
+    const idTemplate = data.id ? `${data.id}` : "";
+
+    template.innerHTML = `
+    <div class="wikidata__header">
+      ${imageTemplate}
+
+      <div>
+        ${tagTypeTemplate}
+        ${titleTemplate}
+        ${countryTemplate}
+      </div>
+    </div>
+
+    <div class="wikidata__content">
+      ${descriptionTemplate}
+
+      <div class="wikidata__footer">
+        <img src="/static/images/wikidata-logo.png" width="34" height="25" alt="Wikidata logo">
+        ${idTemplate}
+      </div>
+    </div>
+  `;
+
+    return template;
+};
+
+/**
+ * createWikidataOverlays
+ *
+ * @param {Object} data
+ */
+const createWikidataOverlays = (data) => {
+    Object.entries(data).forEach(([id, item]) => {
+        // Multiple tags can have the same wikidata entry
+        const overlayContainers = document.querySelectorAll(
+            `[data-js-wikidata-overlay="https://www.wikidata.org/wiki/${id}"]`,
+        );
+
+        overlayContainers.forEach((overlayContainer) => {
```

Comment:
```suggestion
        Array.from(overlayContainers).forEach((overlayContainer) => {
```

Reply:
`forEach` is pretty well supported on `NodeList` now https://caniuse.com/mdn-api_nodelist_foreach, and there are other areas of the project where `forEach` is being used `NodeList` without being converted to an array. For example:

https://github.com/nationalarchives/ds-ohos-wagtail/blob/develop/scripts/src/article.js#L75
https://github.com/nationalarchives/ds-ohos-wagtail/blob/develop/scripts/src/modules/analytics/pictures.js#L7

I'm happy to make the change if you think it's worth it, but it seems superfluous here, unless at some point we want to use an array method that's not available on `NodeList`

Reply:
It's fine - my knowledge is obviously just outdated! 😅